src/cmd/troubleshoot/builder.go

```diff
@@ -88,27 +92,89 @@ func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
 		}
 
 		apiReader := k8scluster.GetAPIReader()
-		tests := []troubleshootFunc{
+		dynakubeUnrelatedTests := []troubleshootFunc{
 			checkNamespace,
-			checkDynakube,
-			checkDTClusterConnection,
-			checkImagePullable,
 		}
+
 		troubleshootCtx := troubleshootContext{
 			context:       context.Background(),
 			apiReader:     apiReader,
 			httpClient:    &http.Client{},
 			namespaceName: namespaceFlagValue,
-			dynakubeName:  dynakubeFlagValue,
 		}
-
-		for _, test := range tests {
+		for _, test := range dynakubeUnrelatedTests {
 			err = test(&troubleshootCtx)
 			if err != nil {
 				logErrorf(err.Error())
 				return nil
 			}
 		}
+		resetLog()
+
+		perDynakubeTests := []troubleshootFunc{
+			checkDynakube,
+			checkDTClusterConnection,
+			checkImagePullable,
+		}
+
+		troubleshootCtx.dynakubeName = dynakubeFlagValue
```

Comment:
i think it would be better if the flag value would be passed to `getDynakubes` as a parameter instead of part of the struct. setting it here just to check it in the function and then re-setting it in the loop seems a bit odd to me

Reply:
And if its no longer used here, couldn't dynakubeName then be removed from the troubleshootContext? Once the troubleshootContext.dynakube is set, it's redundant anyway.